HW1/P4/P4.py

```diff
@@ -0,0 +1,67 @@
+
+# coding: utf-8
+
+# In[1]:
+
+import findspark
+import os
+findspark.init('/home/chongmo/spark') # you need that before import pyspark.
+import pyspark
+from pyspark import SparkContext
+sc =SparkContext()
+
+
+# In[4]:
+
+import ast
+import itertools
+
+
+# In[5]:
+
+#convert the initial RDD (name, issue) into a RDD (issue, {all characters in the issue})
+series_RDD = sc.textFile('source.csv').map(lambda x: ast.literal_eval(x.encode('utf-8').decode('ascii', 'ignore'))[::-1]).groupByKey().mapValues(lambda x: set(x))
```

Comment:
I like this way of parsing each line of the file with ast.literal_eval - saves you from actually mucking around with string parsing.